test/bdd/steps/endpoints.js

```diff
@@ -0,0 +1,202 @@
+    expect(opcode, 'Opcode should only be EQ or IN.').to.satisfy(val => (val === 'EQ' || val === 'IN'));
+    expect(!!this.state[targetNode.toLowerCase()], 'Target node not defined. Use other step to define it.').to.be.equal(true);
+
+
+    const host = this.state[targetNode.toLowerCase()].state.node_rpc_url;
+
+    const jsonQuery = {
+        query:
+            [
+                {
+                    path,
+                    value,
+                    opcode,
+                },
+            ],
+    };
+    const response = await httpApiHelper.apiQueryLocal(host, jsonQuery);
+    this.state.apiQueryLocalResponse = response;
+});
+
+Given(/^I query ([DC|DH|DV]+) node locally for last imported data set id$/, async function (targetNode) {
+    expect(!!this.state.lastImport, 'Nothing was imported. Use other step to do it.').to.be.equal(true);
+    expect(!!this.state.lastImport.data_set_id, 'Last imports data set id seems not defined').to.be.equal(true);
+    expect(targetNode, 'Node type can only be DC, DH or DV.').to.satisfy(val => (val === 'DC' || val === 'DH' || val === 'DV'));
+    expect(!!this.state[targetNode.toLowerCase()], 'Target node not defined. Use other step to define it.').to.be.equal(true);
+
+    const host = this.state[targetNode.toLowerCase()].state.node_rpc_url;
+    const lastDataSetId = this.state.lastImport.data_set_id;
+
+    const response = await httpApiHelper.apiQueryLocalImportByDataSetId(host, lastDataSetId);
+    this.state.apiQueryLocalImportByDataSetIdResponse = response;
+});
+
+Given(/^([DV|DV2]+) publishes query consisting of path: "(\S+)", value: "(\S+)" and opcode: "(\S+)" to the network$/, { timeout: 90000 }, async function (whichDV, path, value, opcode) {
+    expect(!!this.state[whichDV.toLowerCase()], 'DV/DV2 node not defined. Use other step to define it.').to.be.equal(true);
+    expect(opcode, 'Opcode should only be EQ or IN.').to.satisfy(val => (val === 'EQ' || val === 'IN'));
+    const dv = this.state[whichDV.toLowerCase()];
+
+    // TODO find way to pass jsonQuery directly to step definition
+    const jsonQuery = {
+        query:
+            [
+                {
+                    path,
+                    value,
+                    opcode,
+                },
+            ],
+    };
+    const queryNetworkResponse =
+        await httpApiHelper.apiQueryNetwork(dv.state.node_rpc_url, jsonQuery);
+    expect(Object.keys(queryNetworkResponse), 'Reponse should have message and query_id').to.have.members(['message', 'query_id']);
+    expect(queryNetworkResponse.message, 'Message should inform about successful sending of the query').to.be.equal('Query sent successfully.');
+    this.state.lastQueryNetworkId = queryNetworkResponse.query_id;
+    return new Promise((accept, reject) => dv.once('dv-network-query-processed', () => accept()));
+});
+
+Given(/^the ([DV|DV2]+) purchases import from the last query from (a DH|the DC|a DV)$/, function (whichDV, fromWhom, done) {
```

Comment:
'a DH' would mean from any random DH. Think same should apply as for the DC i.e.
```
Given I use 5th node as DH
And the DV purchases import from the last query from the DH
```
or
```
Given I use 5th node as DH4
And the DV purchases import from the last query from the DH4
```

Maybe we should avoid uncertainties.

Reply:
will fix "a" to "the"

Reply:
Let me please address this in some future commit